### PR TITLE
Flatten hierarchy.

### PR DESCRIPTION
**Pull request by ghislainfourny**
Continuing the clean up, this flattens the hierarchy more. PrimaryExpression goes away (but primary expressions are still nicely grouped in a Java package with this name).

### CODE REVIEW

File: src/main/java/org/rumbledb/expressions/operational/CastableExpression.java [hunk 1]
```diff
@@ -22,14 +27,28 @@ public CastableExpression(Expression mainExpression, FlworVarSingleType atomicTy
 
     @Override
     public String serializationString(boolean prefix) {
-        String result = "(castableExpr ";
-        result += this.mainExpression.serializationString(true);
-        result += this.atomicType != null ? " castable as " + this.atomicType.serializationString(prefix) : "";
-        result += ")";
-        return result;
+        StringBuilder result = new StringBuilder();
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/expressions/operational/CastableExpression.java]: Nice

File: src/main/java/org/rumbledb/expressions/primary/PrimaryExpression.java [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
-
-import org.rumbledb.exceptions.ExceptionMetadata;
-import org.rumbledb.expressions.Expression;
-import org.rumbledb.expressions.Node;
-
-import java.util.ArrayList;
-import java.util.List;
-
-
-// TODO(NOT IMPLEMENTED): orderedExpr, unorderedExpr
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/expressions/primary/PrimaryExpression.java]: Would you like to create an issue for this removed TODO: item, or is this already within our development plans?

Comment (ghislainfourny) [reply src/main/java/org/rumbledb/expressions/primary/PrimaryExpression.java]: I have somewhere a masterplan of missing features, i.e., this is under control. Thanks for asking.

**Comment by ghislainfourny** [comment]
Thank you!

**Pull request merged**